internal/surgeon/xray.go

```diff
@@ -0,0 +1,100 @@
+package surgeon
+
+// Library contains raw access to bbolt files for sake of testing or fixing of corrupted files.
+//
+// The library must not be used bbolt btree - just by CLI or tests.
+// It's not optimized for performance.
+
+import (
+	"bytes"
+	"fmt"
+
+	"go.etcd.io/bbolt/internal/guts_cli"
+)
+
+type XRay struct {
+	path string
+}
+
+func NewXRay(path string) XRay {
+	return XRay{path}
+}
+
+func (n XRay) traverseInternal(stack []guts_cli.Pgid, callback func(page *guts_cli.Page, stack []guts_cli.Pgid) error) error {
```

Comment:
nit: you do not expose the function, so it's internal, and it's included in the package `internal/surgeon`.  So no need to add suffix "Internal" anymore.
```suggestion
func (n XRay) traverse(stack []guts_cli.Pgid, callback func(page *guts_cli.Page, stack []guts_cli.Pgid) error) error {
```

Reply:
Done. The idea was to have more convenient (not recursive) wrapper that takes just the root page... 
but finally it was not worth the effort.